training/MESSAGING_OVERVIEW.md

```diff
@@ -0,0 +1,138 @@
+```chapel
+module ExampleModule {
+
+    use ServerConfig;
+    
+    use Reflection;
+    use ServerErrors;
+    use Message;
+
+    use MultiTypeSymbolTable;
+    use MultiTypeSymEntry;
+    use ServerErrorStrings;
+    
+    proc exampleFunction(cmd: string, payload: string, argSize: int, 
+                                    st: borrowed SymTab): MsgTuple throws {
+        var repMsg: string; // response message
+
+        var msgArgs = parseMessageArgs(payload, argSize);
+        var arg1 = msgArgs.getValueOf("arg1");
+        var arg2 = msgArgs.getValueOf("arg2");
+        
+        repMsg = "arg1: %s + arg2: %s".format(arg1, arg2);
+        
+        return new MsgTuple(repMsg, MsgType.NORMAL);
+    }
+    
+    use CommandMap;
+    registerFunction("serverFunction", exampleFunction, getModuleName());
+}
+```
+
+If we were to print the value of `repMsg` on the client after performing this request, the output would be: 
+
+`'arg1: First Argument + arg2: Second Argument'`
+
+<a id="tryIt"></a>
+# Try It!
+
+Now it's your turn to use the information presented in this document to create your own module complete with a request to the Server and response back to the Client!
+
+1. In the root-level directory `arkouda`, create a new python file "mymodule.py"
+   - At a minimum for server communication, you will need to import:
+     - `from typing import cast`
+     - `from arkouda.client import generic_msg`
+2. Create a new function which takes in as many arguments as you want
+   - `def my_request(arg1, arg2, ...):`
+3. Send your arguments to the Server
+   - `repMsg = cast(str, generic_msg(cmd="myServerFunction", args={"arg1": arg1, "arg2": arg2}))`
+4. Print your response from the server
+   - `print(repMsg)`
+5. Using the [example module](#exMod) above as a reference, create a new Chapel module within the root-level `src` directory
+    - This module should include:
+      - A `CommandMap` registered function that your request is being sent into
+      - JSON parsing of your message and some form of manipulation of your arguments. Ex. addition, subtraction, string-reversal, string-concatenation
+      - The creation of a `repMsg` string
+      - The return of a `new MsgTuple` of `MsgType.NORMAL`
+    - **Remember to add your Chapel file name to `ServerModules.cfg`**
+6. To test your new module, you'll first have to use the terminal to navigate to the root-level directory of Arkouda then run `make` to compile your new Chapel code.
+7. Once Chapel compiles, you can then launch the Arkouda server using `./arkouda_server -nl 1`
+8. Using `ipython` you can then easily test your new module
+```python
+import arkouda as ak
+from arkouda.mymodule import my_request
+
+ak.connect() # Connects to the server running locally
+
+my_request("First Argument", 10, ...)
+```
+
+For more examples, visit the [ArkoudaNotebooks](https://github.com/Bears-R-Us/ArkoudaNotebooks) repository.
```

Comment:
`ArkoudaNotebooks` doesn't really have any Client-Server examples, so unless these are being added before this PR, this should be removed.